### PR TITLE
Support `astype()` on BigQuery for basic types

### PR DESCRIPTION
**Pull request by thijs-obj**
Change:
* Support astype() on BigQuery for basic types
* more tests for astype()

### CODE REVIEW

File: bach/bach/series/series.py [hunk 1]
```diff
@@ -228,11 +228,15 @@ def supported_value_to_literal(cls, dialect: Dialect, value: Any) -> Expression:
...
+    def dtype_to_expression(cls, dialect: Dialect, source_dtype: str, expression: Expression) -> Expression:
         """
         INTERNAL: Give the sql expression to convert the given expression, of the given source dtype to the
         dtype of this Series.
-        :return: sql expression
+
+        :param dialect: Database dialect
+        :param source_dtype: dtype of the expression parameter
+        :param expression: expression to cast
+        :return: a new expression that converts the given expression to the dtype of this class
```
Comment (KathiaBarahona) [inline bach/bach/series/series.py:239]: wouldn't it be smth like: "a new expression that casts the given expression to the dialect's db type for the dtype of this class".  I think this currently be misinterpreted IMHO

File: bach/bach/series/series_boolean.py [hunk 2]
```diff
@@ -52,12 +53,18 @@ def supported_value_to_literal(cls, dialect: Dialect, value: bool) -> Expression
...
-    def dtype_to_expression(cls, source_dtype: str, expression: Expression) -> Expression:
+    def dtype_to_expression(cls, dialect: Dialect, source_dtype: str, expression: Expression) -> Expression:
         if source_dtype == 'bool':
             return expression
         if source_dtype not in ['int64', 'string']:
             raise ValueError(f'cannot convert {source_dtype} to bool')
-        return Expression.construct('cast({} as bool)', expression)
+        if is_postgres(dialect):
+            # Postgres cannot directly cast a bigint to bool.
+            # So we do a comparison against 0 (==False) instead
```
Comment (KathiaBarahona) [inline bach/bach/series/series_boolean.py:63]: nice to know :)

File: bach/bach/series/series_numeric.py [hunk 3]
```diff
@@ -224,7 +224,18 @@ def _arithmetic_operation(self, other, operation, fmt_str, other_dtypes=('int64'
         return super()._arithmetic_operation(other, operation, fmt_str, other_dtypes, type_mapping)
 
     def __truediv__(self, other) -> 'Series':
-        return self._arithmetic_operation(other, 'div', 'cast({} as float) / ({})', dtype='float64')
+        # What we do below is essentially the same as: return self.astype('float64') / other
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/bach/series/series_numeric.py:227]: huh I'll take a look

[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/bach/series/series_numeric.py:227]: https://github.com/objectiv/objectiv-analytics/pull/606 this fixes it

[the remark this answers is not in the document]
Comment (thijs-obj) [reply bach/bach/series/series_numeric.py:227]: Fix is here: https://github.com/objectiv/objectiv-analytics/pull/609

**Pull request merged**